plugins/modules/nd_service_instance.py

```diff
@@ -0,0 +1,184 @@
+    choices: [ enable, restart, query, disable, delete ]
+    default: enable
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Enable a Service Instance
+  cisco.nd.nd_service_instance:
+    name: "cisco-terraform"
+    instance_name: "default"
+    target_version: "0.1.15"
+    state: enable
+
+- name: Upgrade a Service Instance
+  cisco.nd.nd_service_instance:
+    name: "cisco-terraform"
+    instance_name: "default"
+    target_version: "0.1.16"
+    state: enable
+
+- name: Restart a Service Instance
+  cisco.nd.nd_service_instance:
+    name: "cisco-terraform"
+    instance_name: "default"
+    state: restart
+
+- name: Query a Service Instance with name and instance_name
+  cisco.nd.nd_service_instance:
+    name: "cisco-terraform"
+    instance_name: "default"
+    state: query
+  register: query_result
+
+- name: Query all Service Instances
+  cisco.nd.nd_service_instance:
+    state: query
+  register: query_results
+
+- name: Disable a Service Instance
+  cisco.nd.nd_service_instance:
+    name: "cisco-terraform"
+    instance_name: "default"
+    state: disable
+
+- name: Delete an existing service instance
+  cisco.nd.nd_service_instance:
+    name: "cisco-terraform"
+    target_version: "0.1.16"
+    state: delete
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        name=dict(type="str", aliases=["service_name"]),
+        instance_name=dict(type="str", aliases=["service_instance_name"]),
+        target_version=dict(type="str"),
+        state=dict(type="str", default="enable", choices=["enable", "query", "disable", "restart", "delete"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "enable", ["name", "instance_name", "target_version"]],
+            ["state", "restart", ["name", "instance_name"]],
+            ["state", "disable", ["name", "instance_name"]],
+            ["state", "delete", ["name", "target_version"]],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    name = nd.params.get("name")
+    target_version = nd.params.get("target_version")
+    instance_name = nd.params.get("instance_name")
+    state = nd.params.get("state")
+
+    base_path = "/nexus/infra/api/firmware/v1"
+    instance_path = "{0}/serviceinstances".format(base_path)
+
+    # To avoid object not found error while querying the object
+    nd.ignore_not_found_error = True
+
+    # Delete part
+    if state == "delete":
+        service_object = nd.query_obj("/nexus/infra/api/firmware/v1/services/{0}:{1}".format(name, target_version))
+        if service_object:
+            nd.existing = service_object
+        else:
+            nd.previous = nd.existing = {}
+            nd.exit_json()
```

Comment:
is it needed that we exit_json in multiple locations in the module, why define for each state the logic that you need and follow flow till the end?